### PR TITLE
NF: rename `Undoable` to `UndoAction`

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
This name is more descriptive of the actual use of this class

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java [hunk 1]
```diff
@@ -1316,14 +1313,15 @@ public void clearUndo() {
 
     /** Undo menu item name, or "" if undo unavailable. */
     @VisibleForTesting
-    public @Nullable Undoable undoType() {
+    public @Nullable
+    UndoAction undoType() {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:1317]: ?

File: AnkiDroid/src/main/java/com/ichi2/libanki/UndoAction.java [hunk 2]
```diff
@@ -35,10 +35,11 @@ public String name(Resources res) {
      * Returned positive integers are card id. Those ids is the card that was discarded and that may be sent back to the reviewer.*/
     public abstract @Nullable Card undo(@NonNull Collection col);
 
-    public static @NonNull Undoable revertToProvidedState (@StringRes int undoNameId, Card card){
+    public static @NonNull
+    UndoAction revertToProvidedState (@StringRes int undoNameId, Card card){
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/UndoAction.java:39]: ?

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Two strange changes, rest looks good

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java (hunk 1 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:1317]: It seems to be a side effect of android studio automated renaming.
Related to a very discussion on discord about new line and annotation

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Pull request merged**